### ISSUE TITLE
MSAL writing null family_id value in JSON string on iOS

### ISSUE THREAD

**Issue by jennyf19** [AzureAD/microsoft-authentication-library-for-dotnet#1176]
**Which Version of MSAL are you using ?**
i think from 3.0.1-preview

**Platform**
xamarin iOS

**Actual behavior**
In the JSON string, we are writing null for the family_id value, it should be string.empty or ""

```
  "home_account_id": "9f44...a8ca",

  "environment": "login.windows.net",

  "client_info": "eyJ1aWQiOiI5Z...RhOGNhIn0",

  "client_id": "4b0db...6f8e0",

  "secret": "OAQ...A9K1dIAA",

  "credential_type": "RefreshToken",

  "family_id": null
```

Am working on a repro of this to provide more context. will update tomorrow.

**Comment by bgavrilMS** [comment]
No need to spend time reproducing this, we probably have unit tests around it. 

According to the JSON spec, null is a valid value and is equivalent to empty string: http://www.json.org/
See also: https://stackoverflow.com/questions/21120999/representing-null-in-json

I do not understand customer impact here.

**Comment by jennyf19** [comment]
@bgavrilMS Here's the issue:
1) We are omitting all the other missing fields, and only writing "null" for family_id
2) Null is valid, but in Obj-C it can cause issues as it's translated to NSNull, which can result in issues for apps that don't except NSNulls.
3) When we did cache compat testing w/native iOS, we were behaving like the other SDKs and not writing null fields to JSON. This change might cause production apps to crash.

**Comment by jmprieur** [comment]
@bgavrilMS : is it the issue you fixed?

**Comment by bgavrilMS** [comment]
Waiting for @henrik-me with instructions on this. PR is out, but not merged.

**Comment by MarkZuber** [comment]
effectively a dup of https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1189 which is fixed.

**Comment by henrik-me** [comment]
Yup, it's a dupe as 1189 is more generic. Should validate this scenario though with the fix for #1189

**Comment by jmprieur** [comment]
this is now fixed and available in MSAL.NET 4.1
